x-pack/plugins/actions/server/config.ts

```diff
@@ -50,6 +50,11 @@ export const configSchema = schema.object({
   rejectUnauthorized: schema.boolean({ defaultValue: true }),
   maxResponseContentLength: schema.byteSize({ defaultValue: '1mb' }),
   responseTimeout: schema.duration({ defaultValue: '60s' }),
+  cleanupFailedExecutionsTask: schema.object({
+    enabled: schema.boolean({ defaultValue: true }),
+    interval: schema.duration({ defaultValue: '15s' }),
```

Comment:
15 seconds?  Is that just so you can test this? :-)

I was thinking more like 12 or 24 hours ... _maybe_ an hour?

Reply:
Yeah, this is a mistake, though I was thinking something small, like `5m`. I'm trying to figure out how long it would take to cleanup environments for customers with `400,000` SOs..

1h processing 100 at a time = ~167 days 🤔 
5m processing 100 at a time = ~14 days

It feels like there are two modes, one to clean up a large amount of data (short interval) and one to clean up nothing or executions that slipped (long interval). In theory, if I fix the task to always be successful, this feature would only be for the former (clean up a large amount of data) and would never be necessary again.

Reply:
Ya, that kinda seems right - long interval that always runs, and a special one that cleans up a large backlog.  Could this be two separate tasks?  Maybe we'd have the large backlog task be disabled, and then enable it if we see > 1000 docs that need to be cleaned.  Once that task went under that threshold (1000 in this case), it would disable itself.

Reply:
>Could this be two separate tasks?

We can return different intervals after each run. I could return, ex `5m` if there are still docs left to clean up after a run.. or return `1h / 12h / 24h` if there are none or something.

Reply:
> We can return different intervals after each run.

That should work out well.  We should probably add a counter for the number of runs to set in alerting health, so we can tell from a report if it appears to be firing too often, or too infrequently.  

I'm wondering if we should log an INFO for the long runs, indicating how many things have been deleted.  Probably overkill - I did notice some debug logging which is probably good enough

Reply:
>That should work out well. We should probably add a counter for the number of runs to set in alerting health, so we can tell from a report if it appears to be firing too often, or too infrequently.

I capture `runs` and `total_cleaned_up` within the task manager state. Do you think that would suffice?

>I'm wondering if we should log an INFO for the long runs, indicating how many things have been deleted. Probably overkill - I did notice some debug logging which is probably good enough

I fear if the user has 400k docs to delete that it would be 4000 info logs (one every 5m) and flood their logs over time.